src/main/java/org/junitpioneer/jupiter/params/IntegralToBytesConverter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.params;
+
+import static java.lang.String.format;
+
+import java.nio.ByteBuffer;
+
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.converter.ArgumentConversionException;
+import org.junit.jupiter.params.converter.ArgumentConverter;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+class IntegralToBytesConverter implements ArgumentConverter {
```

Comment:
In my local attempt, I used a different name for the converter, `ByteArrayArgumentConverter`, which is closer to what I've used for the annotation (see other comment and, again, it's just a proposal and the final choice is yours, of course 🙂 )

Also, what if the converter would extend [`TypedArgumentConverter`](https://junit.org/junit5/docs/5.9.2/api/org.junit.jupiter.params/org/junit/jupiter/params/converter/TypedArgumentConverter.html)? It would get [some basic error handling](https://github.com/junit-team/junit5/blob/bd7d03f517dfca3730fbd123c5fd4d4b70930129/junit-jupiter-params/src/main/java/org/junit/jupiter/params/converter/TypedArgumentConverter.java#L51-L64) out of the box.

```suggestion
class ByteArrayArgumentConverter extends TypedArgumentConverter<Number, byte[]> {
```

AFAIR, JUnit Pioneer tries to stay compatible with older versions of JUnit, and using `TypedArgumentConverter` would require at least JUnit 5.7. I wasn't sure if the version [here](https://github.com/junit-pioneer/junit-pioneer/blob/main/gradle.properties#L1) is the current minimum required by Pioneer. If so, `TypedArgumentConverter` could be an option.